sdk/keyvault/Azure.Security.KeyVault.Secrets/README.md

```diff
@@ -68,110 +68,136 @@ Use the [Azure CLI][azure_cli] snippet below to create/get client secret credent
+
+Console.WriteLine(secret.Name);
+Console.WriteLine(secret.Value);
+Console.WriteLine(secret.Properties.Version);
+Console.WriteLine(secret.Properties.Enabled);
 ```
 
 ### Retrieve a Secret
-`Get` retrieves a secret previously stored in the Key Vault.
+`GetSecretAsync` retrieves a secret previously stored in the Key Vault.
+
+```C# RetrieveSecret
+KeyVaultSecret secret = await client.GetSecretAsync("secret-name");
 
-```C# RetrieveSecret
-KeyVaultSecret secret = client.GetSecret("secret-name");
-
-Console.WriteLine(secret.Name);
-Console.WriteLine(secret.Value);
+Console.WriteLine(secret.Name);
+Console.WriteLine(secret.Value);
 ```
 
 ### Update an existing Secret
-`Update` updates a secret previously stored in the Key Vault. Only the attributes of the secret are updated. To update the value, call `SecretClient.Set` on a `Secret` with the same name.
-
-```C# UpdateSecret
-KeyVaultSecret secret = client.GetSecret("secret-name");
-
-// Clients may specify the content type of a secret to assist in interpreting the secret data when it's retrieved.
-secret.Properties.ContentType = "text/plain";
-
-// You can specify additional application-specific metadata in the form of tags.
-secret.Properties.Tags["foo"] = "updated tag";
-
-SecretProperties updatedSecretProperties = client.UpdateSecretProperties(secret.Properties);
-
-Console.WriteLine(updatedSecretProperties.Name);
-Console.WriteLine(updatedSecretProperties.Version);
-Console.WriteLine(updatedSecretProperties.ContentType);
+`UpdateSecretPropertiesAsync` updates a secret previously stored in the Key Vault. Only the attributes of the secret are updated. To update the value, call `SecretClient.SetSecretAsync` on a secret with the same name.
+
+```C# UpdateSecret
+KeyVaultSecret secret = await client.GetSecretAsync("secret-name");
+
+// Clients may specify the content type of a secret to assist in interpreting the secret data when it's retrieved.
+secret.Properties.ContentType = "text/plain";
+
+// You can specify additional application-specific metadata in the form of tags.
+secret.Properties.Tags["foo"] = "updated tag";
+
+SecretProperties updatedSecretProperties = await client.UpdateSecretPropertiesAsync(secret.Properties);
+
+Console.WriteLine(updatedSecretProperties.Name);
+Console.WriteLine(updatedSecretProperties.Version);
+Console.WriteLine(updatedSecretProperties.ContentType);
 ```
 
 ### Delete a Secret
-`Delete` deletes a secret previously stored in the Key Vault. When [soft-delete][soft_delete] is not enabled for the Key Vault, this operation permanently deletes the secret.
+`StartDeleteSecretAsync` starts a long-running operation to deletes a secret previously stored in the Key Vault. When [soft-delete][soft_delete] is not enabled for the Key Vault, this operation permanently deletes the secret.
+
+```C# DeleteSecret
+DeleteSecretOperation operation = await client.StartDeleteSecretAsync("secret-name");
+
+await operation.WaitForCompletionAsync();
```

Comment:
Same as above I wouldn't demo awaiting the operation unless you're going to purge, and in that case you should have a comment that WaitForCompletion isn't necessary unless you're intending to purge. We need to update all samples and snippets demonstrating Detele, and Recover